functions/Get-DbaAgentAlertCategory.ps1

```diff
@@ -0,0 +1,100 @@
+function Get-DbaAgentAlertCategory {
+    <#
+    .SYNOPSIS
+        Get-DbaAgentAlertCategory retrieves the alert categories.
+
+    .DESCRIPTION
+        Get-DbaAgentAlertCategory makes it possible to retrieve the alert categories.
+
+    .PARAMETER SqlInstance
+        The target SQL Server instance or instances. This can be a collection and receive pipeline input to allow the function to be executed against multiple SQL Server instances.
+
+    .PARAMETER SqlCredential
+        Login to the target instance using alternative credentials. Windows and SQL Authentication supported. Accepts credential objects (Get-Credential)
+
+    .PARAMETER Category
+        The name of the category to filter out. If no category is used all categories will be returned.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Agent, Alert, AlertCategory
+        Author: Patrick Flynn (@sqllensman)
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaAgentAlertCategory
+
+    .EXAMPLE
+        PS C:\> Get-DbaAgentAlertCategory -SqlInstance sql1
+
+        Return all the agent alert categories.
+
+    .EXAMPLE
+        PS C:\> Get-DbaAgentAlertCategory -SqlInstance sql1 -Category 'Severity Alert'
+
+        Return all the agent alert categories that have the name 'Severity Alert'.
+
+    #>
+    [CmdletBinding()]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstance[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Category,
```

Comment:
```suggestion
        [object[]]$Category,
```

@potatoqualitee we used to do these as object types for TEPP, are we still shooting for that?

Reply:
it should be `string`, we generally moved away from `object` after we started doing `InputObject`, tho there are exceptions. we can shoot for tepp but it's not required. it's sort of a pain in the butt so i stopped 😅